comptests/s3/test_rest_api_rgw.py

```diff
@@ -256,16 +333,645 @@ def test_user_create_36665(self):
+        user_name = "tenant11$newtestuser"
+        user_params = {
+            'display-name': display_name,
+            'uid': user_name
+        }
+        self.log.info(
+            "Step 1: Creating a new IAM user with name %s", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Created user details: %s", user_info)
+        self.log.info(
+            "Step 2: Verifying that new IAM user is created successfully")
+        status, user_info = loop.run_until_complete(self.obj.get_user_info(user_params))
+        assert status == HTTPStatus.OK, "Not able to Get user Info. Test Failed"
+        self.log.info("Get user info output: %s", user_info)
+        self.created_users.append(user_params)
+        self.log.info("END : %s", test_case_name)
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-37158')
+    def test_user_create_37158(self):
+        """Test duplicate user specifying tenant in the uid
+        using $ and other using tenant parameter ."""
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("START: Test duplicate user specifying tenant in the uid using $ and"
+                      " other using tenant parameter .")
+        user_name = f"{self.user_name_prefix}1"
+        tenant = "tnt1"
+        user_name_with_tenant = tenant + '$' + user_name
+        user_params = {
+            'display-name': user_name,
+            'uid': user_name,
+            'tenant': tenant
+        }
+        user_params2 = {
+            'display-name': user_name,
+            'uid': user_name_with_tenant
+        }
+        self.log.info(
+            "Step 1: Creating a new IAM user with name %s", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Created user details: %s", user_info)
+        self.log.info(
+            "Step 2: Verifying that new IAM user is created successfully")
+        status, user_info = loop.run_until_complete(self.obj.get_user_info(user_params))
+        assert status == HTTPStatus.OK, "Not able to Get user Info. Test Failed"
+        self.log.info("Get user info output: %s", user_info)
+        self.created_users.append(user_params)
+        self.log.info(
+            "Step 3: Creating another IAM user with same name "
+            "containing tenant name as %s", str(user_name_with_tenant))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params2))
+        assert status == HTTPStatus.CONFLICT, "Haven't got the expected failure. Test Failed"
```

Comment:
Message in such case would be same like duplicate user names. We may want to check the consistency of the message as well.